2018-07-17 15:33:00,477 ERROR: Exception on /edit_profile [GET] [in /home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/home/mani/work/next upload/basic-webapp/app/main/routes.py", line 85, in edit_profile
    return render_template('edit_profile.html', title='Edit Profile', form=form)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/templating.py", line 135, in render_template
    context, ctx.app)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/templating.py", line 117, in _render
    rv = template.render(context)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/home/mani/work/next upload/basic-webapp/app/templates/edit_profile.html", line 29, in top-level template code
    <a class="nav-link" href="{{ url_for('main.login') }}">Home <span class="sr-only">(current)</span></a>
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/helpers.py", line 356, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 2061, in handle_url_build_error
    reraise(exc_type, exc_value, tb)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/helpers.py", line 345, in url_for
    force_external=external)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/werkzeug/routing.py", line 1776, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'main.login'. Did you mean 'auth.login' instead?
2018-07-17 15:33:27,962 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:33:27,965 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:33:27,965 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:34:58,892 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:34:58,895 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:34:58,895 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:35:23,172 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:35:23,175 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:35:23,175 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:35:47,285 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:35:47,287 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:35:47,287 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:41:21,126 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:41:21,129 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:41:21,129 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:42:32,658 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:42:32,661 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:42:32,661 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:46:41,674 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:46:41,678 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:46:41,678 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:47:48,553 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:47:48,556 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:47:48,556 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:48:43,601 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:48:43,604 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:48:43,604 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:49:26,607 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:49:26,610 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:49:26,610 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:57:38,457 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:57:38,468 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 15:57:38,468 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 16:04:45,135 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 16:04:45,138 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 16:04:45,138 INFO: M.S.Blog startup [in /home/mani/work/next upload/basic-webapp/app/__init__.py:77]
2018-07-17 16:07:49,922 ERROR: Exception on /follow/manas [GET] [in /home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py:1761]
Traceback (most recent call last):
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/mani/work/next upload/venv/lib/python3.6/site-packages/flask_login/utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "/home/mani/work/next upload/basic-webapp/app/main/routes.py", line 100, in follow
    flash('You are following %(username)s!', username=username)
TypeError: flash() got an unexpected keyword argument 'username'
